README.md

```diff
@@ -92,26 +92,75 @@ Two specific points to expand on:
 
 ### Setting up
 
-1. Follow the [Flutter installation guide](https://docs.flutter.dev/get-started/install)
-   for your platform of choice.
-2. Switch to the latest version of Flutter by running `flutter channel main`
-   and `flutter upgrade` (see [Flutter version](#flutter-version) below).
-3. Ensure Flutter is correctly configured by running `flutter doctor`.
-4. Start the app with `flutter run`, or from your IDE.
+1. Setup the pinned version of flutter.
+
+    1.1. Make sure you have initialized submodule by doing:
+
+    ```sh
+    # Update and initialize submodule
+    git submodule update --init
+    # Run setup-vendor-flutter
+    tools/setup-vendor-flutter
+    ```
```

Comment:
Similar to https://github.com/zulip/zulip-flutter/pull/579#discussion_r1540348954:
> Let's try to write this message with the audience in mind of someone who's just cloned this repo and is starting to go through the README, and has never heard of Git submodules (and even if they have, doesn't know our particular layout or how we're using submodules).

So here:
```suggestion
    1.1. Initialize the pinned Flutter tree:

    ```sh
    git submodule update --init
    tools/setup-vendor-flutter
    ```
```

Reply:
Relatedly: can we move the `git submodule update --init` inside the script, so that this step is just a single command?

Hmm, though on the other hand, and zooming out a bit: what happens when as a contributor you rebase (or reset to a new `origin/main`), and the version that's pinned has changed? You'll need to run some `git submodule` command to update your local clone, right?

If so, that seems like something we should mention in this doc too. But it also points to keeping an explicit `git submodule` step here in the setup after all, to avoid having something that's initially magic but where the developer later needs to look behind the magic anyway.

Reply:
Ah,yes, in my latest changes I've explored moving `git submodule update --init` into the `setup-vendor-flutter` script and making this script runnable any/all the time. If it's already set up, it'll be a noop.